sources/sqlserver/data_test.go

```diff
@@ -0,0 +1,299 @@
+	assert.Equal(t, []spannerData{{table: tableName, cols: cols, vals: []interface{}{float64(4.2), int64(6), "prisoner zero"}}}, rows)
+}
+
+func TestConvertData(t *testing.T) {
+
+	numStr := "33753785954695469456094506940.33333333982343435"
+	numVal := new(big.Rat)
+	numVal.SetString(numStr)
+
+	singleColTests := []struct {
+		name  string
+		ty    ddl.Type
+		srcTy string      // Source DB type
+		in    string      // Input value for conversion.
+		e     interface{} // Expected result.
+	}{
+		{"bit 0", ddl.Type{Name: ddl.Bool}, "", "0", false},
+		{"bit 1", ddl.Type{Name: ddl.Bool}, "", "1", true},
+		{"bytes", ddl.Type{Name: ddl.Bytes, Len: ddl.MaxLength}, "", string([]byte{137, 80}), []byte{0x89, 0x50}}, // need some other approach to testblob type
+		{"date", ddl.Type{Name: ddl.Date}, "", "2019-10-29", getDate("2019-10-29")},
+		{"float64", ddl.Type{Name: ddl.Float64}, "", "42.6", float64(42.6)},
+		{"int64", ddl.Type{Name: ddl.Int64}, "", "42", int64(42)},
+		{"string", ddl.Type{Name: ddl.String, Len: ddl.MaxLength}, "", "eh", "eh"},
+		{"datetime", ddl.Type{Name: ddl.Timestamp}, "datetime", "2019-10-29 05:30:00", getTimeWithoutTimezone(t, "2019-10-29 05:30:00")},
+		{"decimal", ddl.Type{Name: ddl.Numeric}, "decimal", "234.90909090909", big.NewRat(23490909090909, 100000000000)},
+		{"numeric", ddl.Type{Name: ddl.Numeric}, "numeric", numStr, numVal},
+	}
+	tableName := "testtable"
+	for _, tc := range singleColTests {
+		col := "a"
+		conv := buildConv(
+			ddl.CreateTable{
+				Name:     tableName,
+				ColNames: []string{col},
+				ColDefs:  map[string]ddl.ColumnDef{col: {Name: col, T: tc.ty, NotNull: false}},
+				Pks:      []ddl.IndexKey{}},
+			schema.Table{Name: tableName, ColNames: []string{col}, ColDefs: map[string]schema.Column{col: {Type: schema.Type{Name: tc.srcTy}}}})
+		conv.TimezoneOffset = "+05:30"
+		t.Run(tc.in, func(t *testing.T) {
+			at, ac, av, err := ConvertData(conv, tableName, []string{col}, conv.SrcSchema[tableName], tableName, []string{col}, conv.SpSchema[tableName], []string{tc.in})
+			checkResults(t, at, ac, av, err, tableName, []string{col}, []interface{}{tc.e}, tc.name)
+		})
+	}
+}
+
+func TestConvertMultiColData(t *testing.T) {
+	multiColTests := []struct {
+		name  string
+		cols  []string      // Input columns.
+		vals  []string      // Input values.
+		ecols []string      // Expected columns.
+		evals []interface{} // Expected values.
+	}{
+		{
+			name:  "Cols in order",
+			cols:  []string{"a", "b", "c"},
+			vals:  []string{"6", "6.6", "1"},
+			ecols: []string{"a", "b", "c"},
+			evals: []interface{}{int64(6), float64(6.6), true},
+		},
+		{
+			name:  "Cols out of order",
+			cols:  []string{"b", "c", "a"},
+			vals:  []string{"6.6", "1", "6"},
+			ecols: []string{"b", "c", "a"},
+			evals: []interface{}{float64(6.6), true, int64(6)},
+		},
+		{
+			name:  "Null column",
+			cols:  []string{"a", "b", "c"},
+			vals:  []string{"6", "<nil>", "1"},
+			ecols: []string{"a", "c"},
```

Comment:
This should also change since for nil values, we plan on going with `"NULL"` instead of `"<nil>"` as per the valsToStrings function.

Reply:
Correct, updated with "NULL"